lib/sycamore/sycamore/tests/unit/query/execution/test_sycamore_executor.py

```diff
@@ -102,3 +104,55 @@ def test_run_plan_with_caching(test_count_docs_query_plan, mock_sycamore_docsetr
             # No new directories should have been created.
             existing_dirs = [os.path.join(temp_dir, x) for x in os.listdir(temp_dir)]
             assert set(existing_dirs) == set(cache_dirs)
+
+
+def test_run_summarize_data_plan(mock_sycamore_docsetreader):
+
+    plan = LogicalPlan(
+        query="Test query",
+        result_node=1,
+        nodes={
+            0: QueryDatabase(node_id=0, description="Load data", index="test_index"),
+            1: SummarizeData(node_id=1, description="Summarize data", question="Summarize this data", inputs=[0]),
+        },
+    )
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        with (
+            patch("sycamore.reader.DocSetReader", new=mock_sycamore_docsetreader),
+            patch("sycamore.tests.unit.query.conftest.MOCK_SCAN_NUM_DOCUMENTS", new=1000),
+        ):
+            context = sycamore.init(
+                params={
+                    "default": {"llm": Mock(spec=LLM)},
+                    "opensearch": {
+                        "os_client_args": {
+                            "hosts": [{"host": "localhost", "port": 9200}],
+                            "http_compress": True,
+                            "http_auth": ("admin", "admin"),
+                            "use_ssl": True,
+                            "verify_certs": False,
+                            "ssl_assert_hostname": False,
+                            "ssl_show_warn": False,
+                            "timeout": 120,
+                        }
+                    },
+                }
+            )
+
+            # First run should populate cache.
+            executor = SycamoreExecutor(context, cache_dir=temp_dir)
```

Comment:
Is this test fast?  If it is, then fine to leave in unit tests, but if it's more than 5-10s, I'd like to get it moved into integration tests.  I initially thought this would use ray (which basically guarantees it's slow), but I'm no longer sure.